CMakeLists.txt

```diff
@@ -223,11 +223,25 @@ if (NOT CMAKE_BUILD_TYPE_UC STREQUAL "RELEASE")
     endif ()
 endif()
 
+if (CMAKE_BUILD_TYPE_UC STREQUAL "RELEASE"
+    OR CMAKE_BUILD_TYPE_UC STREQUAL "RELWITHDEBINFO"
+    OR CMAKE_BUILD_TYPE_UC STREQUAL "MINSIZEREL")
+    set (STRIP_HEAVY_DEBUG_SYMBOLS_DEFAULT ON)
+else()
+    set (STRIP_HEAVY_DEBUG_SYMBOLS_DEFAULT OFF)
+endif()
+# Provides faster linking and lower binary size.
+# Tradeoff is the inability to debug some source files with e.g. gdb
+# (empty stack frames and no local variables)."
+option(STRIP_HEAVY_DEBUG_SYMBOLS
```

Comment:
AFAIR it is auto generated, but #37441 removes it

Reply:
This cmake-in-clickhouse article looks strange, and I'm pretty sure that it is out of date, if not now, then one day it definitely will without auto-generation. So I'm voting to remove it altogether.

Reply:
Also lines in this document already outdated, even with auto generation it should uses hashes.

Reply:
- #38684

Reply:
Agree, the article is strange and I too forgot to update it in my past commits. I also concur to remove it as it creates overhead in keeping it in-sync with the build system options.

That aside, I suggest to not squeeze the removal into this PR, especially since other people may have an opinion on the removal as well. After all, the article is at least a good entry point for developers to discover the available build options and we should at least check that the most useful options (e.g. ENABLE_LIBRARIES=1 or USE_STATIC_LIBRARIES=0) are mentioned somewhere else, e.g. in a "tips and tricks" section in the developer guide.

Reply:
>the article is at least a good entry point for developers to discover the available build options

`cmake -LAH` can be used, but some refactoring is required to make it useful, for example add a prefix for all important variables.

Also I still think that it can provide more harm if some important options will be renamed, after all developer can look at sources, there is not too much cmake code in clickhouse. But don't get me wrong, I'm not forcing removing this documentation, just a suggestion, let's see what other developers think.